File: cs/auth/1.public/auth/model/client_info.cs
using System.Diagnostics.CodeAnalysis;

namespace HyperId.SDK.Authorization
{
    /// <summary>
    /// struct eClientInfo
    /// </summary>
    public struct ClientInfo
    {
           public ClientInfo(
            [NotNull] string clientId,
            [NotNull] string redirectUri,
            [NotNull] AuthMethod authorizationMethod,
            [AllowNull] string? clientSecret,
            [AllowNull] string? rsaKeysPem)
        {
            ClientId = clientId;
            AuthMethod = authorizationMethod;
            RedirectUri = redirectUri;
            ClientSecret = clientSecret;
            RSAKeysPem = rsaKeysPem;
        }
        /// <summary>
        /// IsValid
        /// </summary>
        public readonly bool IsValid()
        {
            if(string.IsNullOrEmpty(ClientId) || string.IsNullOrEmpty(RedirectUri))
            {
                return false;
            }

            if(AuthMethod == AuthMethod.CLIENT_SECRET_BASIC
                || AuthMethod == AuthMethod.CLIENT_SECRET_HMAC)
            {
                return !string.IsNullOrEmpty(ClientSecret);
            }
            else
            {
                return !string.IsNullOrEmpty(RSAKeysPem);
            }
        }

        [NotNull]
        public string ClientId { get; private set; }

        [NotNull]
        public AuthMethod AuthMethod { get; private set; }

        [NotNull]
        public string RedirectUri { get; private set; }

        [AllowNull]
        public string? ClientSecret { get; private set; }

        [AllowNull]
        public string? RSAKeysPem { get; private set; }
    }
}//namespace  HyperId.SDK.Authorization

File: android/hyperid_sdk/src/main/java/ai/hyper_id/sdk/api/auth/model/ClientInfo.kt
package ai.hyper_id.sdk.api.auth.model

import java.security.interfaces.RSAPrivateKey

//==================================================================================================
//	eClientInfo
//--------------------------------------------------------------------------------------------------
data class ClientInfo(val clientId				: String,
					  val redirectUri			: String,
					  val authorizationMethod	: AuthorizationMethod,
					  val clientSecret			: String?,
					  val privateRSAKey			: RSAPrivateKey?)
{
	//==================================================================================================
	//	isValid
	//--------------------------------------------------------------------------------------------------
	fun isValid() : Boolean
	{
		return	clientId.isNotBlank()
			&&	redirectUri.isNotBlank()
			&&	when(authorizationMethod)
				{
					AuthorizationMethod.CLIENT_SECRET,
					AuthorizationMethod.CLIENT_SECRET_HS256 -> !clientSecret.isNullOrBlank()
					AuthorizationMethod.CLIENT_RS256        -> privateRSAKey != null
				}
	}
	//==================================================================================================
	//	toString
	//--------------------------------------------------------------------------------------------------
	override fun toString() : String
	{
		return "[ClientInfo clientId/$clientId, " +
			   "redirectUri/$redirectUri, " +
			   "authorizationMethod/$authorizationMethod, " +
			   "clientSecret/$clientSecret, ]"
	}
}
